README.md
# gradient_toolbar

[![pub package](https://img.shields.io/pub/v/gradient_toolbar.svg)](https://pub.dartlang.org/packages/gradient_toolbar)


A Flutter package to add a gradient toolbar with a lot of customization.

### Screenshots

<img src="Screenshot1.png" height="500em" /> <img src="Screenshot2.png" height="500em" /> <img src="Screenshot3.png" height="500em" />

## Usage
To use this plugin, add `gradient_toolbar` as a dependency in your [pubspec.yaml](https://github.com/rohitgupta1694/FlutterGithubUserSearch/blob/master/pubspec.yaml) file.

```yaml
  dependencies:
    flutter:
      sdk: flutter
    gradient_toolbar: ^0.1.0
```


### Example

Please find the [example](https://github.com/rohitgupta1694/GradientToolBar/blob/master/example/lib/main.dart):

``` dart

class ExampleApp extends StatefulWidget {
  @override
  _ExampleAppState createState() => _ExampleAppState();
}

class _ExampleAppState extends State<ExampleApp> {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      backgroundColor: Colors.white,
      /// Define Column in body and then add GradientToolBar widget as a child
      body: Column(
        children: <Widget>[
        ///This is the Gradient tool bar widget
          GradientToolBar(
            title: "Gradient Toolbar", // Toolbar Title Text
            textColor: Colors.white, // Toolbar Title Text Color
            fontSize: 22.0, // Toolbar Title Text Font Size
            fontStyle: FontStyle.normal, // Toolbar Title Text Font Style
            fontWeight: FontWeight.w600, // Toolbar Title Text Font Weight
            needLeadingButton: true, // Toolbar Back Button Visibility Flag
            needTrailingButton: true, // Toolbar Trailing Button Visibility Flag
            trailingIcon: Icons.search, // Toolbar Trailing Button Icon
            leadingIconColor: Colors.white, // Toolbar Leading Button Color
            trailingIconColor: Colors.white, // Toolbar Trailing Button Color
            onLeadingIconClick: _handleLeadingButtonClick, // Toolbar Leading Button Callback
            onTrailingIconClick: _handleTrailingButtonClick, // Toolbar Trailing Button Callback
            toolbarStartColor: Color(0xFF414345), // Toolbar Gradient Start Color
            toolbarEndColor: Color(0xFF232526), // Toolbar Gradient End Color
            gradientTileMode: TileMode.clamp, // Toolbar Gradient Tile Mode
          ),
          // Use Expanded widget to use rest of the screen.
          Expanded(
              child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[Text("Gradient Toolbar")],
          ))
        ],
      ),
    );
  }

  void _handleLeadingButtonClick(bool clickStatus) {
    //TODO: Do something
  }

  void _handleTrailingButtonClick(bool clickStatus) {
    //TODO: Do something
  }
}


```

### Created & Maintained By

[Rohit Gupta](https://github.com/rohitgupta1694)

* ([Twitter@rohitgupta1694](https://www.twitter.com/rohitgupta1694))
* ([Medium@rohitgupta1694](https://medium.com/@rohitgupta1694))

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.io/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.io/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.1.0]
* Final release

## [0.0.4]
* Updated CHANGELOG.md file

## [0.0.3]
* Updated README.md file

## [0.0.2]
* Fixed Nullable attributes bugs

## [0.0.1]
* First Release




example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:gradient_toolbar/gradient_toolbar.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: ExampleApp(),
    );
  }
}

class ExampleApp extends StatefulWidget {
  @override
  _ExampleAppState createState() => _ExampleAppState();
}

class _ExampleAppState extends State<ExampleApp> {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      backgroundColor: Colors.white,

      /// Define Column in body and then add GradientToolBar widget as a child
      body: Column(
        children: <Widget>[

          ///This is the Gradient tool bar widget
          GradientToolBar(
            title: "Gradient ToolBar",
            needLeadingButton: false,
            needTrailingButton: false,
          ),

          /// Use Expanded widget to use rest of the screen.
          Expanded(
              child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
                children: <Widget>[Text("Gradient Toolbar sample")],
          ))
        ],
      ),
    );
  }

  void _handleLeadingButtonClick(bool clickStatus) {
    //TODO: Do something
  }

  void _handleTrailingButtonClick(bool clickStatus) {
    //TODO: Do something
  }
}


lib/gradient_toolbar.dart
library gradient_toolbar;

import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

class GradientToolBar extends StatelessWidget {
  final double barHeight = 50.0;

  GradientToolBar(
      {Key key,
      this.title,
        this.textColor = Colors.white,
        this.fontSize = 20.0,
        this.fontStyle = FontStyle.normal,
        this.fontWeight = FontWeight.w600,
        this.toolbarStartColor = const Color(0xFF414345),
        this.toolbarEndColor = const Color(0xFF232526),
        this.gradientTileMode = TileMode.clamp,
      this.needLeadingButton,
      this.needTrailingButton,
        this.trailingIcon = Icons.search,
      this.onLeadingIconClick,
      this.onTrailingIconClick,
        this.leadingIconColor = Colors.white,
        this.trailingIconColor = Colors.white})
      : assert(title != null),
        assert(needLeadingButton != null),
        assert(needTrailingButton != null),
        super(key: key);

  final String title;
  final Color textColor,
      toolbarStartColor,
      toolbarEndColor,
      leadingIconColor,
      trailingIconColor;
  final double fontSize;
  final FontStyle fontStyle;
  final FontWeight fontWeight;
  final IconData trailingIcon;
  final TileMode gradientTileMode;
  final bool needLeadingButton, needTrailingButton;
  final ValueChanged<bool> onLeadingIconClick, onTrailingIconClick;

  void _onTrailingIconClick() {
    if (onTrailingIconClick != null)
      onTrailingIconClick(true);
    else
      print(
          "Please pass a trailing icon callback when initializing GradientToolbar");
  }

  void _onLeadingButtonClick() {
    if (onLeadingIconClick != null)
      onLeadingIconClick(true);
    else
      print(
          "Please pass a leading icon callback when initializing GradientToolbar");
  }

  @override
  Widget build(BuildContext context) {
    SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light);
    final double statusBarHeight = MediaQuery.of(context).padding.top;
    return Container(
      padding: new EdgeInsets.only(top: statusBarHeight),
      height: statusBarHeight + barHeight,
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: <Widget>[
          getCustomToolBarLeadingWidget(context),
          getCustomToolBarTextWidget(),
          getCustomToolBarTrailingWidget(),
        ],
      ),
      decoration: new BoxDecoration(
        boxShadow: [
          new BoxShadow(
            spreadRadius: 5.0,
            color: Colors.grey,
            offset: new Offset(0.0, 1.0),
            blurRadius: 8.0,
          )
        ],
        gradient: new LinearGradient(
            colors: [toolbarStartColor, toolbarEndColor],
            begin: const FractionalOffset(0.0, 0.0),
            end: const FractionalOffset(0.5, 0.0),
            stops: [0.0, 1.0],
            tileMode: gradientTileMode),
      ),
    );
  }

  getTextWidget(String title) => Text(title,
      style: TextStyle(
          fontSize: fontSize,
          color: textColor,
          fontWeight: fontWeight,
          fontStyle: fontStyle));

  getBackButton(BuildContext context) => IconButton(
        icon: Icon(defaultTargetPlatform != TargetPlatform.iOS
            ? Icons.arrow_back
            : Icons.arrow_back_ios),
        color: leadingIconColor,
        onPressed: _onLeadingButtonClick,
      );

  getCustomToolBarLeadingWidget(BuildContext context) =>
      needLeadingButton ? getBackButton(context) : Container(child: Text(""));

  Widget getCustomToolBarTextWidget() {
    if (!needLeadingButton && needTrailingButton) {
      return Padding(
        padding: const EdgeInsets.only(left: 48.0),
        child: getTextWidget(title),
      );
    } else if (needLeadingButton && !needTrailingButton) {
      return Padding(
        padding: const EdgeInsets.only(right: 48.0),
        child: getTextWidget(title),
      );
    } else {
      return getTextWidget(title);
    }
  }

  getCustomToolBarTrailingWidget() => needTrailingButton
      ? IconButton(
          icon: Icon(trailingIcon),
          color: trailingIconColor,
          onPressed: _onTrailingIconClick, // null disables the button
        )
      : Container(child: Text(""));
}


lib/example.dart
import 'package:flutter/material.dart';
import 'package:gradient_toolbar/gradient_toolbar.dart';

class ExampleApp extends StatefulWidget {
  @override
  _ExampleAppState createState() => _ExampleAppState();
}

class _ExampleAppState extends State<ExampleApp> {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      backgroundColor: Colors.white,
      /// Define Column in body and then add GradientToolBar widget as a child
      body: Column(
        children: <Widget>[
          new GradientToolBar(
            title: "Gradient Toolbar", // Toolbar Title Text
            textColor: Colors.white, // Toolbar Title Text Color
            fontSize: 22.0, // Toolbar Title Text Font Size
            fontStyle: FontStyle.normal, // Toolbar Title Text Font Style
            fontWeight: FontWeight.w600, // Toolbar Title Text Font Weight
            needLeadingButton: true, // Toolbar Back Button Visibility Flag
            needTrailingButton: true, // Toolbar Trailing Button Visibility Flag
            trailingIcon: Icons.search, // Toolbar Trailing Button Icon
            leadingIconColor: Colors.white, // Toolbar Leading Button Color
            trailingIconColor: Colors.white, // Toolbar Trailing Button Color
            onLeadingIconClick: _handleLeadingButtonClick, // Toolbar Leading Button Callback
            onTrailingIconClick: _handleTrailingButtonClick, // Toolbar Trailing Button Callback
            toolbarStartColor: Color(0xFF414345), // Toolbar Gradient Start Color
            toolbarEndColor: Color(0xFF232526), // Toolbar Gradient End Color
            gradientTileMode: TileMode.clamp, // Toolbar Gradient Tile Mode
          ),
          /// Use Expanded widget to use rest of the screen.
          Expanded(
              child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[Text("Gradient Toolbar")],
          ))
        ],
      ),
    );
  }

  void _handleLeadingButtonClick(bool clickStatus) {
    //TODO: Do something
  }

  void _handleTrailingButtonClick(bool clickStatus) {
    //TODO: Do something
  }
}
